src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2981,8 +2986,276 @@ void Schedule::invalidNamePattern( const std::string& namePattern,  std::size_t
+            well.updateConnections( std::move(well_connections) );
+        }
+
+        if (!segments.empty()) {
+            std::vector<Segment> segments_list;
+            for (const auto& segment_pair : segments)
+                segments_list.push_back( std::move(segment_pair.second) );
+
+            auto comp_pressure_drop = WellSegments::CompPressureDrop::HFA;
+            std::shared_ptr<Opm::WellSegments> well_segments = std::make_shared<Opm::WellSegments>(comp_pressure_drop, segments_list);
+            well.updateSegments( std::move(well_segments) );
+        }
+
+        this->addWell(well, report_step);
+        this->addWellToGroup(well.groupName(), well.name(), report_step);
+    }
+
+    m_tuning.update(report_step, rst_state.tuning);
+    m_events.addEvent( ScheduleEvents::TUNING_CHANGE , report_step);
+}
+
+namespace {
+/*
+  The insane trickery here (thank you Stackoverflow!) is to be able to provide a
+  simple templated comparison function
+
+     template <typename T>
+     int not_equal(const T& arg1, const T& arg2, const std::string& msg);
+
+  which will print arg1 and arg2 on stderr *if* T supports operator<<, otherwise
+  it will just print the typename of T.
+*/
+
+
+template<typename T, typename = int>
+struct cmpx
 {
+    int neq(const T& arg1, const T& arg2, const std::string& msg) {
+        if (arg1 == arg2)
+            return 0;
+
+        std::cerr << "Error when comparing <" << typeid(arg1).name() << ">: " << msg << std::endl;
+        return 1;
+    }
+};
+
+template <typename T>
+struct cmpx<T, decltype(std::cout << T(), 0)>
+{
+    int neq(const T& arg1, const T& arg2, const std::string& msg) {
+        if (arg1 == arg2)
+            return 0;
+
+        std::cerr << "Error when comparing: " << msg << " " << arg1 << " != " << arg2 << std::endl;
+        return 1;
+    }
+};
+
+
+template <typename T>
+int not_equal(const T& arg1, const T& arg2, const std::string& msg) {
+    return cmpx<T>().neq(arg1, arg2, msg);
+}
+
+
+template <>
+int not_equal(const double& arg1, const double& arg2, const std::string& msg) {
+    if (Opm::cmp::scalar_equal(arg1, arg2))
+        return 0;
+
+    std::cerr << "Error when comparing: " << msg << " " << arg1 << " != " << arg2 << std::endl;
+    return 1;
+}
+
+template <>
+int not_equal(const UDAValue& arg1, const UDAValue& arg2, const std::string& msg) {
+    if (arg1.is<double>())
+        return not_equal( arg1.get<double>(), arg2.get<double>(), msg);
+    else
+        return not_equal( arg1.get<std::string>(), arg2.get<std::string>(), msg);
+}
+
+
+std::string well_msg(const std::string& well, const std::string& msg) {
+    return "Well: " + well + " " + msg;
+}
+
+
+}
+
+bool Schedule::cmp(const Schedule& sched1, const Schedule& sched2, std::size_t report_step) {
+    int count = 0;
+    for (const auto& wname : sched1.wellNames(report_step)) {
```

Comment:
Is there an implicit check that
```C++
sched1.wellNames(report_step) == sched2.wellNames(report_step)
```
before we call this function?  If not, I think that would be an appropriate additional check here.

Reply:
I agree - added